File: 13-04-2026/digital_rotation_escape.py
def get_rotation(n):
    nLength, arr = len(str(n)), list(str(n))
    
    for i in range(nLength):
        num = int("".join(arr))
        if num % nLength == 0:
            return i
        arr.append(arr.pop(0))

    return "none"

print(get_rotation(123)) # 0.
print(get_rotation(13579)) # 3.
print(get_rotation(24681)) # "none".
print(get_rotation(84138789345)) # 6.

File: 13-04-2026/digital_rotation_escape.js
function getRotation(n) {
  let [nLength, arr] = [`${n}`.length, `${n}`.split("")];

  for (let i = 0; i < nLength; i++) {
    let num = parseInt(arr.join(""));
    if (num % nLength === 0) return i;
    arr.push(arr.shift());
  }

  return "none";
}

console.log(getRotation(123)); // 0.
console.log(getRotation(13579)); // 3.
console.log(getRotation(24681)); // "none".
console.log(getRotation(84138789345)); // 6.
